src/main/javascript/service/message-service.js

```diff
@@ -1,10 +1,22 @@
 'use strict';
 
+angular.module('cat.service.message', [
+    'cat.config.messages'
+])
+
 /**
  * @ngdoc service
  * @name cat.service.message:$globalMessages
  */
-angular.module('cat.service.message', []).service('$globalMessages', function CatGlobalMessages($rootScope) {
+    .service('$globalMessages', function CatGlobalMessages($rootScope) {
+
+    function Message(data) {
+        data = data || {};
+        this.text = data ? data.text : '';
+        this.type = data ? data.type : undefined;
+        this.timeToLive = data ? data.timeToLive : 0;
+    }
+
```

Comment:
this can be simplified as you ensure that data is defined:

```
function Message(data) {
    data = data || {}:
    this.text = data.text || '';
    this.type = data.type;
    this.timeToLive = data.timeToLive || 0;
}
```